common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -228,7 +228,7 @@ private synchronized boolean connect() throws IOException {
         String localHostName = NetworkAddressUtils.getLocalHostName(mTachyonConf);
         LOG.info("Trying to get local worker host : " + localHostName);
         workerNetAddress = mMasterClient.user_getWorker(false, localHostName);
-        mIsLocal = true;
+        mIsLocal = NetworkAddressUtils.getFqdnHost(workerNetAddress).equals(localHostName);
```

Comment:
The line 242 guarantee a remote worker because in case we have the MaxFreeBalancer, we will get a worker at line 230 and in case we have the LocalFirstBalancer, we will return the local worker at line 230 if there is one or a random remote worker at line 242 otherwise.
This part will be changed when removing the random parameters. I will reorganize the code at this time.

I changed the way of matching the 2 addresses.